res/schema.xml

```diff
@@ -446,4 +446,37 @@ METADATA
       );
     </sql>
   </revision>
+  <revision version="29" min_compatible="3">
+    <description>
+      Add tables to support nested crates.
+    </description>
+    <sql>
+      CREATE TABLE IF NOT EXISTS crateClosure (
+        parentId INTEGER,
+        childId INTEGER,
+        depth INTEGER
+      );
+
+      CREATE TABLE IF NOT EXISTS cratePath (
+        crateId INTEGER,
+        idPath TEXT,
+        namePath TEXT
+      );
+
+      ALTER TABLE crates RENAME TO old_crates;
```

Comment:
Please add a comment about removing the unique constraint from the name column. The SQL code does not reveal this fact.